src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbolBase.cs

```diff
@@ -630,7 +641,7 @@ protected bool IsAutoProperty
         /// Backing field for automatically implemented property, or
         /// for a property with an initializer.
         /// </summary>
-        internal SynthesizedBackingFieldSymbol BackingField { get; }
+        internal SynthesizedBackingFieldSymbol BackingField { get; set; }
```

Comment:
> We should review all places that read this property and make appropriate adjustments, if necessary. 

Has this been done?

Reply:
The interesting places where we read the property:

- [x] In `InitializerSemanticModel.BindEqualsValue` tested with `TestGetSpeculativeSemanticModelInAutoPropInitializer1` in `SemanticModelAPITests.cs`.
- [x] In `SyntaxTreeSemanticModel.CreateMemberModel` covered by the same test.
- [ ] In `MethodBodySynthesizer`, I think it's unreachable for field-keywords. But adding a PROTOTYPE to re-validate later.
- [ ] `AbstractFlowPass`.
- [ ] `DefiniteAssignment`
- [ ] `LocalRewriter`

Reply:
Regarding ````AbstractFlowPass```.
It looks like the logic there requires an adjustment since the behavior of ```IsAutoPropertyWithGetAccessor``` has changed. We can no longer guarantee that accessing the property under the checked conditions is always equivalent to accessing a backing field. For example, if the corresponding accessor has a body, we don't really know what it is doing.

Reply:
@AlekseyTs Do I need to record a PROTOTYPE for this or start working in this area?

Reply:
> Do I need to record a PROTOTYPE for this or start working in this area? I'm more with a PROTOTYPE since this depends on AccessingAutoPropertyFromConstructor, which isn't yet adjusted. What do you think?

Given the scope of the initial PR, there is nothing to do here now. A prototype comment for the future wouldn't hurt, I guess.